lib/services/nodes-api-service.js

```diff
@@ -249,18 +249,36 @@ function nodeApiServiceFactory(
         return Promise.try(function() {
             query = waterline.nodes.find(query);
 
-            if (options.skip) query.skip(options.skip);
-            if (options.limit) query.limit(options.limit);
+            if (options.skip) {
+                query.skip(options.skip);
+            }
+            if (options.limit) {
+                query.limit(options.limit);
+            }
 
             return query.populate('obms');
         });
     };
 
     NodeApiService.prototype.postNode = function(body) {
-        return Promise.resolve().then(function() {
+        return Promise.try (function() {
+            assert.string(body.type, 'node type is required');
+            if (_.values(Constants.NodeTypes).indexOf(body.type) < 0) {
+                throw new Error('Unknown node type "' + body.type + '"');
+            }
+            assert.optionalArrayOfObject(
+                body.obmSettings, 'obmSettings should be an array of object');
+        })
+        .then(function() {
             return waterline.nodes.create(body);
         }).tap(function(node) {
             return eventsProtocol.publishNodeEvent(node, 'added');
+        }).tap(function(node) {
```

Comment:
@srinia6: I examined the schema for `POST /nodes` (see https://github.com/RackHD/on-http/blob/master/static/monorail-2.0.yaml#L83), it doesn't tell how to specify the obm when manually adding a node.
And by looking at the rest code of `postNode` function, you will it still use the `obmSettings` (see https://github.com/yyscamper/on-http/blob/b31f6b199676172de8ddb37c0504de6df1cac197/lib/services/nodes-api-service.js#L308).

Anyway, we always need a way to let user specify the obm while manually adding a node, previously the obm document is bind with node, so it's pretty straightforward, now the obm is moved out, the situation is a little hard. If let user manually add the obm before creating node, but you don't know the node id; If bind obm document with node, then there is some what mismatch with our internal structure.

So @srinia6 @brianparry : What's your preference for how to specify the obm while adding a node?